### PR TITLE
Refactor: gateway api optimization

### PR DESCRIPTION
**Pull request by njucjc**
1. rename some ambiguous field of `Gateway`.
2. generate node information list in `Gateway` status.
3. get node PrivateIP Automatically.

@BSWANG @DrmagicE

### CODE REVIEW

File: pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go [hunk 1]
```diff
@@ -41,29 +41,35 @@ type GatewaySpec struct {
 	// NodeSelector is a label query over nodes that managed by the gateway.
 	// The nodes in the same gateway should share same layer 3 network.
 	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
-	// Backend is the VPN tunnel implementation, only "libreswan" is supported currently.
+	// VPNDriver is the VPN tunnel implementation, only "libreswan" is supported currently.
 	// The default value "libreswan" will be used if empty.
-	Backend string `json:"backend"`
+	VPNDriver string `json:"vpnDriver,omitempty"`
```
Comment (DrmagicE) [inline pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go]: Actually, I am doubting whether the VPN driver should be configured in the CRD.
This field implies that the VPN driver can be switched on the fly (without restarting the raven pod), which I think is a tricky task and is not that necessary.

I propose to remove this field, and design a configuration file or flag in raven.
@BSWANG any thoughts?

Comment (BSWANG) [reply pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go]: `VPN Driver field` allows gateways use different backend when some edge's network block some VPN protocol. But as present, I approve remove the `VPNDriver` field, using global environment on agent to config `VPNDriver`.

Comment (njucjc) [reply pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go]: done.

**Comment by BSWANG** [comment]
LGTM

**Pull request merged**